perllib/FixMyStreet/App/Controller/Waste.pm

```diff
@@ -1129,7 +1130,14 @@ sub add_report : Private {
 
     $c->forward('setup_categories_and_bodies') unless $c->stash->{contacts};
     $c->forward('/report/new/non_map_creation', [['/waste/remove_name_errors']]) or return;
-    $c->forward('/report/new/redirect_or_confirm_creation');
+    if ( $no_confirm ) {
+        my $report = $c->stash->{report};
+        $report->state('unconfirmed');
```

Comment:
This is presumably for if you are logged in, or the report was created by a staff member and so already confirmed by this point? Do you want to leave a comment about that?

Reply:
No, this is because we don't want to confirm a report that we've not processed the payment for, so yes, I should totally add a comment.

Reply:
I suspect there's some more work to be done to confirm email addresses elsewhere but with this I just want to make it work as is.

Reply:
Yes, sorry, that's what I meant - it would only have already been marked confirmed by this point if they were already logged in or a staff member.